cpp/tests/experimental/join/quadtree_point_in_polygon_test_large.cu

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuspatial_test/base_fixture.hpp>
+#include <cuspatial_test/random.cuh>
+#include <cuspatial_test/test_util.cuh>
+#include <cuspatial_test/vector_equality.hpp>
+#include <cuspatial_test/vector_factories.cuh>
+
+#include <cuspatial/error.hpp>
+#include <cuspatial/experimental/geometry/box.hpp>
+#include <cuspatial/experimental/point_in_polygon.cuh>
+#include <cuspatial/experimental/polygon_bounding_boxes.cuh>
+#include <cuspatial/experimental/spatial_join.cuh>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/device_uvector.hpp>
+#include <rmm/exec_policy.hpp>
+#include <rmm/mr/device/device_memory_resource.hpp>
+
+#include <rmm/mr/device/per_device_resource.hpp>
+#include <thrust/gather.h>
+#include <thrust/sort.h>
+
+/*
+ * The test uses the same quadtree structure as in pip_refine_test_small. However, the number of
+ * randomly generated points under all quadrants (min_size) are increased to be more than the
+ * number of threads per-block.
+ */
+
+template <typename T>
+struct PIPRefineTestLarge : public cuspatial::test::BaseFixture {
+};
+
+using TestTypes = ::testing::Types<float, double>;
+
+TYPED_TEST_CASE(PIPRefineTestLarge, TestTypes);
+
+template <typename T>
+inline auto generate_points(
```

Comment:
I'm actually working on a multipoint generator in `geometry_generator.cuh` in https://github.com/rapidsai/cuspatial/pull/1002

https://github.com/rapidsai/cuspatial/blob/2b46f18d5c144c0af4bc2c6e5f5be8ee2c66a72e/cpp/include/cuspatial_test/geometry_generator.cuh#L284

Reply:
Thanks! Can use in a followup.